src/components/GuestCard.js

```diff
@@ -27,57 +31,93 @@ export default function GuestCard({ personalData }) {
 
   return (
     <StyledArticle>
-      <StyledCheckButton type="button" onClick={() => toggleSelected(id)}>
-        {selected === true ? (
-          <img src={checkedCircle} alt="selected" />
-        ) : (
-          <img src={emptyCircle} alt="unselected" />
-        )}
-      </StyledCheckButton>
-      <p>{name}</p>
-      <StyledDetailsButton onClick={() => navigate(`/details/${id}`)}>
-        <img src={detailIcon} alt="details" />
-      </StyledDetailsButton>
+      <StyledBasicSection>
+        <StyledCheckButton type="button" onClick={() => toggleSelected(id)}>
+          {selected ? (
+            <img src={checkedCircle} alt="selected" />
+          ) : (
+            <img src={emptyCircle} alt="unselected" />
+          )}
+        </StyledCheckButton>
+        <StyledName>{name}</StyledName>
+        <StyledDetailsButton
+          onClick={() => toggleShowGuestInfo(!showGuestInfo)}
+        >
+          <img
+            src={detailIcon}
+            alt="details"
+            style={{ transform: showGuestInfo ? "" : "rotate(90deg)" }}
+          />
+        </StyledDetailsButton>
+      </StyledBasicSection>
+      <StyledGuestInfoSection
+        style={{ display: showGuestInfo ? "unset" : "none" }}
+      >
+        <StyledInfoP>
+          {intolerances.length > 0 ? "Food should be:" : ""}
```

Comment:
Per convention you should use null for not rendering instead of empty strings. You can also use the logical and operator: `intolerances.length > 0 && "Food should be:"`